### DISCUSSION TITLE
doris能用WITH函数写递归查询吗

### DISCUSSION THREAD

**Discussion by AndLlk** [apache/doris#28410] [A - General / Q&A] [answered]
查询SQL
WITH  tree  AS ( 
 SELECT MOID, SrcMRP_ID, SrcMRP_PID, CAST(CONCAT(MODocNO,'|',ItemInfo_ItemCode) AS VARCHAR(600)) AS chain
 FROM dwd_QTL_MRP_MOPlanOrder A1 WHERE  left(A1.ItemInfo_ItemCode,1) in ('F')
 UNION ALL
 SELECT pt.MOID, pt.SrcMRP_ID, pt.SrcMRP_PID, CONCAT(pt2.chain,'->',pt.MODocNO,'|',pt.ItemInfo_ItemCode) FROM dwd_QTL_MRP_MOPlanOrder pt join tree pt2
 on pt.SrcMRP_PID=pt2.SrcMRP_ID
)
SELECT * FROM tree;

**Comment by LemonLiTree** [comment, marked as answer]
不支持递归

**Reply by AndLlk** [reply]
感谢大佬
